crawl_tool/crawl.py

```diff
@@ -110,6 +110,8 @@ def crawl(source, url_list, download, wait_time, verbose, output_folder):
         links = soup.find_all("a", href=True)
         if download:
             print(f"-------Crawling on url {url}-------\n")
+            if not os.path.isdir(output_address):
+                os.makedirs(output_address)
```

Comment:
`os.makedirs(output_address, exist_ok=True)` is sufficient. See https://docs.python.org/3/library/os.html#os.makedirs

Reply:
Code updated.